source/common/http/conn_pool_grid.cc

```diff
@@ -201,17 +225,22 @@ ConnectionPool::Cancellable* ConnectivityGrid::newStream(Http::ResponseDecoder&
   if (pools_.empty()) {
     createNextPool();
   }
-
+  PoolIterator pool = pools_.begin();
+  if (is_http3_broken_) {
+    ENVOY_LOG(trace, "HTTP/3 is broken to host '{}', skipping.", describePool(**pool),
+              host_->hostname());
+    createNextPool();
```

Comment:
do we need this?  AFIK we can't mark H3 broken until we've already created both pools.

Reply:
It's probably not strictly needed. But since the method is a no-op if both pools are already created, it felt a bit "safer" to ensure that we created to pool before increment the pool iterator.  (I could also imagine some future usage in which the Cluster has some a priori knowledge which it uses to mark HTTP/3 broken when creating a ConnectivityGrid).

But I can remove it if you feel strongly.

Reply:
how about a comment about it likely not being necessary?

Reply:
Done!